swap/swap.go

```diff
@@ -80,19 +82,46 @@ func NewParams() *Params {
 }
 
 // New - swap constructor
-func New(stateStore state.Store, prvkey *ecdsa.PrivateKey, backend contract.Backend) *Swap {
+func new(stateStore state.Store, prvkey *ecdsa.PrivateKey, backend contract.Backend, disconnectThreshold uint64, paymentThreshold uint64) *Swap {
 	return &Swap{
 		store:               stateStore,
 		peers:               make(map[enode.ID]*Peer),
 		backend:             backend,
 		owner:               createOwner(prvkey),
 		params:              NewParams(),
-		paymentThreshold:    DefaultPaymentThreshold,
-		disconnectThreshold: DefaultDisconnectThreshold,
-		oracle:              NewPriceOracle(),
+		disconnectThreshold: disconnectThreshold,
+		paymentThreshold:    paymentThreshold,
+		honeyPriceOracle:    NewHoneyPriceOracle(),
 	}
 }
 
+// NewSWAP - swap constructor with integrity checks
+func NewSWAP(dbPath string, prvkey *ecdsa.PrivateKey, backendURL string, disconnectThreshold uint64, paymentThreshold uint64) (*Swap, error) {
```

Comment:
I am a bit torn with this naming. Usually, indeed acronyms are all capitalized: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

However, we never use this format in the code so far.

Furthermore, constructors usually don't contain the package name, as users of it would use it like this: `swap.NewSWAP`, which repeats it https://golang.org/doc/effective_go.html#names

Unfortunately this has the disadvantage that in this case, we'd have `New` and `new`, because we need `new` as well for tests.

But I still think I prefer this rather than `NewSWAP`

Reply:
Changed to `New`